### PR TITLE
added debug code and handled the worker disconnect case

### PR DESCRIPTION
**Pull request by coder-guy22296**

### CODE REVIEW

**Review by seanwessmith — CHANGES_REQUESTED** [review]
we don't accept new connections in the launch_simulation.c function

File: networking/Dispatcher/launch_simulation.c [hunk 1]
```diff
@@ -55,8 +56,17 @@ static void	*handle_worker_connection(void *input)
 			make_new_event_thread(params->dispatcher, worker->next);
 		}
 		msg = get_worker_msg(cur_worker);
+		printf("msg status: %d\n", msg.error);
 		if (msg.error == -1)
 			printf("get worker message failed with err %d\n", errno);
+		else if (msg.error == 0)
```
Comment (seanwessmith) [inline networking/Dispatcher/launch_simulation.c:62]: disconnect pthread[worker->tid]
free worker in linked list

**Review by seanwessmith — APPROVED** [review]

File: networking/Dispatcher/launch_simulation.c [hunk 2]
```diff
@@ -88,5 +98,6 @@ void		launch_simulation(t_dispatcher *dispatcher)
 	cur_worker->tid = calloc(1, sizeof(pthread_t));
 	printf("ALLOCCED\n");
 	pthread_create(cur_worker->tid, NULL, handle_worker_connection, param);
+	sleep(999999);
```
Comment (riking) [inline networking/Dispatcher/launch_simulation.c:101]: you can sleep forever by locking a mutex twice

Comment (seanwessmith) [reply networking/Dispatcher/launch_simulation.c:101]: Thanks Kane, I'll replace the sleep with mutex

**Pull request merged**